processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -860,7 +864,15 @@ public Number get()
                     @Override
                     public ColumnCapabilities getColumnCapabilities(String columnName)
                     {
-                      return getColumnCapabilites(index, columnName);
+                      ColumnCapabilities capabilities = getColumnCapabilites(index, columnName);
+                      if (capabilities == null && !virtualColumns.isEmpty()) {
```

Comment:
much of the virtual column capability creation logic is shared across the QueryableIndexStorageAdapter and IncrementalIndexStorageAdapter, maybe this could be a function